php/class-utils.php

```diff
@@ -543,7 +543,7 @@ public static function pathinfo( $path, $flags = 15 ) {
 	 * @return bool
 	 */
 	public static function looks_like_json( $thing ) {
-		return ! empty( $thing ) && is_string( $thing ) && in_array( ltrim( $thing )[0], array( '{', '[' ), true );
+		return is_string( $thing ) && ! empty( ltrim( $thing ) ) && in_array( $thing[0], array( '{', '[' ), true );
```

Comment:
<!-- highlight something positive -->
**👏 praise:**

@epigeyre thank you for the PR. However, it will fail to test if the JSON will look like this ` {}`.

Also, I'm realizing that the cognitive load for this method is a bit too much, making it hard to read.

I'm adding a suggestion here. Would you mind to give it a try?

```suggestion
		if ( ! is_string( $thing ) ) {
			return false;
		}

		$thing = trim( $thing );
	
		if ( empty( $thing ) ) {
			return false;
		}
	
		if ( ! in_array( $thing[0], array( '{', '[' ), true ) ) {
			return false;
		}

		return true;
```

Note that in your approach, you're not trimming the string before the last check, and that's why it will return `false` on the test I've made.

I'll be happy to add it to the upcoming release.

Reply:
@pereirinha You are right this is much better. Thanks !